applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_broadcast.c

```diff
@@ -46,6 +46,11 @@ static void scan_restart_worker(struct k_work *work)
 {
 	int ret;
 
+	ret = bt_le_scan_stop();
+	if (ret) {
+		LOG_WRN("Stop scan failed: %d", ret);
+	}
```

Comment:
Since we are starting scan right after, is there a reason why we stop and restart scanning? 

i.e. could this be removed together with 
```c
ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_BROADCAST, NULL);
```

Reply:
I discussed this with the team.
The reason why we do bt_mgmt_scan_start() here is to re-add the scan callbacks. And that is why scanning is stopped first.

The way we now register and unregister the scan callbacks and use scan_cb_registered for "checking if we are scanning or not" might not be optimal. This could potentially be improved, maybe in this PR? @alexsven @koffes @rick1082 

I will be leaving it as is, for now at least.